src/exec/run-batch-submitter.ts

```diff
@@ -151,26 +151,30 @@ export const run = async () => {
     func: () => Promise<TransactionReceipt>
   ): Promise<void> => {
     // Clear all pending transactions
-    const pendingTxs = await sequencerSigner.getTransactionCount('pending')
-    const latestTxs = await sequencerSigner.getTransactionCount('latest')
-    if (pendingTxs > latestTxs) {
-      log.info('Detected pending transactions. Clearing all transactions!')
-      for (let i = latestTxs; i < pendingTxs; i++) {
-        const response = await sequencerSigner.sendTransaction({
-          to: await sequencerSigner.getAddress(),
-          value: 0,
-          nonce: i,
-        })
-        log.info(`Submitting transaction with nonce: ${i}; hash: ${response.hash}`)
-        await BatchSubmitter.getReceiptWithResubmission(
-          response,
-          [],
-          sequencerSigner,
-          parseInt(requiredEnvVars.NUM_CONFIRMATIONS, 10),
-          60 * 1_000,  // Attempt resubmission every 60 seconds
-          log,
-        )
+    try {
+      const pendingTxs = await sequencerSigner.getTransactionCount('pending')
+      const latestTxs = await sequencerSigner.getTransactionCount('latest')
+      if (pendingTxs > latestTxs) {
+        log.info('Detected pending transactions. Clearing all transactions!')
+        for (let i = latestTxs; i < pendingTxs; i++) {
+          const response = await sequencerSigner.sendTransaction({
+            to: await sequencerSigner.getAddress(),
+            value: 0,
+            nonce: i,
+          })
+          log.info(`Submitting transaction with nonce: ${i}; hash: ${response.hash}`)
+          await BatchSubmitter.getReceiptWithResubmission(
+            response,
+            [],
+            sequencerSigner,
+            parseInt(requiredEnvVars.NUM_CONFIRMATIONS, 10),
+            60 * 1_000,  // Attempt resubmission every 60 seconds
+            log,
+          )
+        }
       }
+    } catch (err) {
+      log.error('Cannot clear transactions', err)
```

Comment:
Agreed this should be in a flag but this looks good!

One hesitation that I have here is that if we can't clear transactions, should we actually `exit(1)`? Do we want to run the batch submitter if we are unable to clear?

Reply:
Good call on exiting the process if failing to clear the txs, added this code behind a config flag